rpcs3/Emu/Cell/lv2/sys_game.cpp

```diff
@@ -3,14 +3,78 @@
 #include "Emu/Memory/vm_ptr.h"
 #include "Emu/Cell/ErrorCodes.h"
 #include "Emu/System.h"
-
+#include "Emu/system_utils.hpp"
 #include "Emu/IdManager.h"
 #include "Utilities/Thread.h"
 
 #include "sys_game.h"
 
 LOG_CHANNEL(sys_game);
 
+struct board_storage
+{
+public:
+	bool read(void* buffer)
+	{
+		if (!buffer)
+			return false;
+
+		storage.atomic_op([&](be_t<u128>& data)
+		{
+			memcpy(buffer, &data, size);
+		});
+
+		return true;
+	}
+
+	bool write(void* buffer)
+	{
+		if (!buffer)
+			return false;
+
+		storage.atomic_op([&](be_t<u128>& data)
+		{
+			memcpy(&data, buffer, size);
+		});
+
+		written = true;
+
+		return true;
+	}
+
+	board_storage()
+	{
+		if (fs::file file; file.open(file_path, fs::read))
+			file.read(&storage.raw(), size);
+		else
+			memset(&storage.raw(), 0, size);
+	}
+
+	board_storage(const board_storage&) = delete;
+
+	board_storage& operator =(const board_storage&) = delete;
+
+	~board_storage()
+	{
+		if (written)
+		{
+			if (fs::file file; file.open(file_path, fs::create + fs::write + fs::lock))
+			{
+				file.seek(0, fs::seek_set);
+				file.write(&storage.raw(), size);
+				file.trunc(size);
+			}
+		}
+		g_fxo->get<board_storage>(); // Register destructor
```

Comment:
Oh, this was needed for RAWSPU manager because nowhere else g_fxo->get was insterted, you only need to access the type once for it to register, welcome to c++ magic.

Reply:
Removed.